atomicapp/nulecule/main.py

```diff
@@ -317,7 +319,9 @@ def run(self, cli_provider, answers_output, ask,
         if not self.answers_file:
             self._process_answers()
 
-        self.nulecule.load_config(config=self.nulecule.config, ask=ask)
+        self.nulecule.load_config(ask=ask)
+        if cli_provider:
+            self.nulecule.config.set('provider', cli_provider)
         self.nulecule.render(cli_provider, dryrun)
         self.nulecule.run(cli_provider, dryrun)
```

Comment:
changing above to something like this can help -> `self.nulecule.run(self.nulecule.config.set('provider', cli_provider), dryrun)`
If we do it here it removes the requirement of making checks at all here: https://github.com/projectatomic/atomicapp/pull/720/files#diff-093821a6e6f49d13d49fdcb4f911695dR93